prediction_market_agent/agents/microchain_agent/deploy.py

```diff
@@ -33,13 +36,21 @@
 GENERAL_AGENT_TAG = "general_agent"
 
 
-class DeployableMicrochainAgent(DeployableAgent):
+class DeployableMicrochainAgentAbstract(DeployableAgent, metaclass=abc.ABCMeta):
     model = SupportedModel.gpt_4o
-    n_iterations = 50
-    load_historical_prompt: bool = False
-    system_prompt_choice: SystemPromptChoice = SystemPromptChoice.TRADING_AGENT
-    identifier: AgentIdentifier = AgentIdentifier.MICROCHAIN_AGENT_OMEN
-    description: str
+    max_iterations: int | None = 50
+    import_actions_from_memory = 0
```

Comment:
If we are running forever and ever, what happens if the pod is restarted for some reason? If `import_actions_from_memory > 0`, then last N actions that the agent has in the LongTermMemory will be feeded into the history list.